ros_buildfarm/templates/misc/rosdistro_cache_job.xml.em

```diff
@@ -79,6 +79,9 @@
         ' --cidfile=$WORKSPACE/docker_generate_rosdistro_cache/docker.cid' +
         ' --net=host' +
         ' -v $WORKSPACE/rosdistro_cache:/tmp/rosdistro_cache' +
+        ' -v $HOME/.ssh/known_hosts:/etc/ssh/ssh_known_hosts:ro' +
+        ' -v $SSH_AUTH_SOCK:/tmp/ssh_auth_sock' +
+        ' -e SSH_AUTH_SOCK=/tmp/ssh_auth_sock' +
```

Comment:
SSH_AUTH_SOCK is only ever unset if the SSH Agent plugin isn't installed. I made commits to the buildfarm_deployment repo to install that plugin, but then I discovered when I tried to deploy a new buildfarm that the plugin wasn't getting installed. It turned out that I had to explicitly install all the dependencies for the plugin as well as the plugin itself, so I updated the repo with the following commit: https://github.com/clearpathrobotics/buildfarm_deployment/commit/80aefe38887874b2fd67f8e8647d677beca84d7a

Please try checking out that version of buildfarm_deployment (at the top of reconfigure.bash in buildfarm_deployment_config you can specify the URL and branch/version to use) with that commit, reconfiguring your master, and trying again.

You've raised a good point though, in that anyone who deploys a buildfarm with buildfarm_deployment before my changes and then updates ros_buildfarm with the changes from this PR will have broken jobs until they either re-run reconfigure.bash or otherwise cause the SSH Agent plugin to be installed.